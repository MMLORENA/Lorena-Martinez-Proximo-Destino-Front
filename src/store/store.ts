import { Action, configureStore, ThunkAction } from "@reduxjs/toolkit";
import { destinationsReducer } from "./reducer/destinationsSlice/destinationsSlice";
import { UIReducer } from "./reducer/uiSlice";
import { userReducer } from "./reducer/userSlice";

export const store = configureStore({
  reducer: {
    user: userReducer,
    ui: UIReducer,
    destinations: destinationsReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
