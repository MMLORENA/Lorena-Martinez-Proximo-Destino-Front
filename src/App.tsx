import "@fontsource/montserrat";
import { ThemeProvider } from "styled-components";
import { Route, Routes, Navigate } from "react-router-dom";
import styledMainTheme from "./styledMainTheme";
import AppStyled from "./AppStyled";
import RegisterPage from "./pages/RegisterPage/RegisterPage";
import Modal from "./components/Modal/Modal";
import { useAppSelector } from "./store/hooks";
import PageNotFound from "./pages/PageNotFound/PageNotFound";
import LoginPage from "./pages/LoginPage/LoginPage";
import NavigationMenu from "./components/NavigationMenu/NavigationMenu";
import FeedbackModal from "./components/FeedbackModal/FeedbackModal";
import DestinationsPage from "./pages/DestinationsPage/DestinationsPage";

const App = () => {
  const {
    modal: { isModalOpen, modalType, modalText },
    feedback: { isFeedbackOpen, feedbackText, feedbackType },
  } = useAppSelector((state) => state.ui);

  return (
    <ThemeProvider theme={styledMainTheme}>
      <AppStyled>
        <NavigationMenu />
        {isFeedbackOpen && (
          <FeedbackModal type={feedbackType} text={feedbackText} />
        )}
        {isModalOpen && <Modal type={modalType} text={modalText} />}
        <Routes>
          <Route path="/" element={<Navigate to="/login" />} />
          <Route path="/login" element={<LoginPage />} />
          <Route path="/registro" element={<RegisterPage />} />
          <Route path="/destinos" element={<DestinationsPage />} />
          <Route path="*" element={<PageNotFound />} />
        </Routes>
      </AppStyled>
    </ThemeProvider>
  );
};

export default App;
