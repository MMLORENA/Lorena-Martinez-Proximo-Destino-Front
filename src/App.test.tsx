import { BrowserRouter } from "react-router-dom";
import App from "./App";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "./store/store";

const mockSelectorReturn = {
  modal: {
    isModalOpen: true,
    modalText: "Algo ha ido mal",
    modalType: "error",
  },
  feedback: {
    isFeedbackOpen: true,
    feedbackText: "Mar",
    feedbackType: "welcome",
  },
};

jest.mock("./store/hooks", () => ({
  ...jest.requireActual("./store/hooks"),
  useAppSelector: () => mockSelectorReturn,
}));

describe("Given an App component", () => {
  describe("When isModalOpen is true", () => {
    test("Then it should render the modal component with a plane-icon inside", async () => {
      render(
        <Provider store={store}>
          <BrowserRouter>
            <App />
          </BrowserRouter>
        </Provider>
      );
      const modalText = screen.getByTestId("icon-plane");

      expect(modalText).toBeInTheDocument();
    });
  });

  describe("When isFeedbackOpen is true", () => {
    test("Then it should render the feedback component with a '¡Hola!' text inside", async () => {
      const expectedSalutation = "¡Hola!";
      render(
        <Provider store={store}>
          <BrowserRouter>
            <App />
          </BrowserRouter>
        </Provider>
      );

      const feedbackSalutation = screen.getByText(expectedSalutation);

      expect(feedbackSalutation).toHaveTextContent(expectedSalutation);
    });
  });
});
