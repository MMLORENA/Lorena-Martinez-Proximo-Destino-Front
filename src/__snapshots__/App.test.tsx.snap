// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an App component When it's render Then it shoul render an App Component 1`] = `
<div
  className="sc-bczRLJ jVieLS"
>
  <div
    className="sc-papXJ ejrLZq"
  >
    <div
      className="feedback__container"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-circle-xmark feedback__icon-cross"
        data-icon="circle-xmark"
        data-prefix="fas"
        data-testid="icon-cross"
        focusable="false"
        onClick={[Function]}
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM175 175c9.4-9.4 24.6-9.4 33.9 0l47 47 47-47c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9l-47 47 47 47c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-47-47-47 47c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l47-47-47-47c-9.4-9.4-9.4-24.6 0-33.9z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span
        className="feedback__text"
      >
        Â¡Hola
      </span>
      <span
        className="feedback__text"
      >
        Mar
      </span>
      <span
        className="feedback__emoji"
      >
        ðŸ¤—
      </span>
    </div>
  </div>
  <div
    className="sc-jSMfEi iVbdgA modal"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-xmark modal__icon-cross"
      data-icon="xmark"
      data-prefix="fas"
      data-testid="icon-cross"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-plane-slash modal__icon-plane"
      data-icon="plane-slash"
      data-prefix="fas"
      data-testid="icon-plane"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 640 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M38.8 5.1C28.4-3.1 13.3-1.2 5.1 9.2S-1.2 34.7 9.2 42.9l592 464c10.4 8.2 25.5 6.3 33.7-4.1s6.3-25.5-4.1-33.7L440.6 320h73.8c34.2 0 93.7-28 93.7-64c0-35-59.5-64-93.7-64l-116.6 0L297.2 16.1C291.5 6.2 280.9 0 269.4 0H213.2c-10.6 0-18.3 10.2-15.4 20.4l40.3 140.9L38.8 5.1zm2.7 123.6C36 130.6 32 135.9 32 142c0 1.3 .2 2.6 .5 3.9L64 256 32.5 366.1c-.4 1.3-.5 2.6-.5 3.9c0 7.8 6.3 14 14 14H88c5 0 9.8-2.4 12.8-6.4L144 320H246.9l-49 171.6c-2.9 10.2 4.8 20.4 15.4 20.4l56.2 0c11.5 0 22.1-6.2 27.8-16.1l65.3-114.3L41.5 128.7z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="modal__text"
    >
      Algo ha ido mal
    </span>
  </div>
</div>
`;
